Fix even more compiler warnings
Author: Henrik Fehlauer <rkflx@lab12.net>

Index: synctex/synctex_parser.c
===================================================================
--- synctex.orig/synctex_parser.c
+++ synctex/synctex_parser.c
@@ -529,8 +529,8 @@ typedef struct synctex_reader_t {
     char * current; /*  current location in the buffer */
     char * start;   /*  start of the buffer */
     char * end;     /*  end of the buffer */
-    int min_size;
-    int size;
+    size_t min_size;
+    size_t size;
     int lastv;
     int line_number;
     SYNCTEX_DECLARE_CHAR_OFFSET
@@ -759,7 +759,6 @@ synctex_reader_p synctex_reader_init_wit
             (char *)_synctex_malloc(reader->size+1); /*  one more character for null termination */
         if (NULL == reader->start) {
             _synctex_error("!  malloc error in synctex_reader_init_with_output_file.");
-        bailey:
 #ifdef SYNCTEX_DEBUG
             return reader;
 #else
@@ -5293,7 +5292,7 @@ content_loop:
 #       pragma mark + SCAN KERN
 #   endif
             ns = _synctex_parse_new_kern(scanner);
-        continue_scan:
+        /* continue_scan: */
             if (ns.status == SYNCTEX_STATUS_OK) {
                 if (child) {
                     _synctex_node_set_sibling(child,ns.node);
@@ -6531,7 +6530,7 @@ SYNCTEX_INLINE static synctex_point_s _s
     return old;
 }
 SYNCTEX_INLINE static synctex_box_s _synctex_data_box(synctex_node_p node) {
-    synctex_box_s box = {0,0,0,0};
+    synctex_box_s box = {{0,0},{0,0}};
     int n;
     n = synctex_node_width(node);
     if (n<0) {
@@ -6547,7 +6546,7 @@ SYNCTEX_INLINE static synctex_box_s _syn
     return box;
 }
 SYNCTEX_INLINE static synctex_box_s _synctex_data_xob(synctex_node_p node) {
-    synctex_box_s box = {0,0,0,0};
+    synctex_box_s box = {{0,0},{0,0}};
     int n;
     n = synctex_node_width(node);
     if (n>0) {
@@ -6563,7 +6562,7 @@ SYNCTEX_INLINE static synctex_box_s _syn
     return box;
 }
 SYNCTEX_INLINE static synctex_box_s _synctex_data_box_V(synctex_node_p node) {
-    synctex_box_s box = {0,0,0,0};
+    synctex_box_s box = {{0,0},{0,0}};
     int n;
     n = _synctex_node_width_V(node);
     if (n<0) {
